### PR TITLE
[gherkin-c] Make gherkin-c up to date

### PR DESCRIPTION
**Pull request by brasmusson**
## Summary

Update gherkin-c from v7.0.4 standard to v24.0.x standard

## Details

Remaining work:
* Add a plugin interface to the id generation
Rather that having gherkin-c to know how to generated UUID:s, let the client of gherkin-c provide such a component. Thereby gherkin-c become less complex or have fewer dependencies. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.

### CODE REVIEW

**Review by aurelien-reeves — COMMENTED** [review]
I have not enough recent experience to do a proper, secure review of C code :'(

Regarding the build, and anything else, it looks fine

File: gherkin/c/README.md [hunk 1]
```diff
@@ -1,4 +1,115 @@
...
+The source file [`gherkin_cli.c`](src/gherkin_cli.c) can be used as an example of using the Gherkin library.
+
+The Gherkin library contains only an incrementing id generator (generating the ids "1", "2", ...), should UUIDs be necessary, the user needs to provide an implementation of the id_generator.h interface that generates UUID strings.
+
+# Building
+
+Three artifacts can be built:
+
+* the Gherkin library for static linking (libgherkin.a)
+* the Gherkin library for dynamic linking (libgherkin.so.&lt;version>)
```
[the remark this answers is not in the document]
Comment (brasmusson) [reply gherkin/c/README.md:74]: If using "`.so.<version>)`" then what is actually rendered is ".so.)", to using "`.so.&lt;version>)`" is necessary to get ".so.&lt;version>) actually rendered.

File: gherkin/c/include/error_event.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#ifndef GHERKIN_ERROR_EVENT_H_
```
[the remark this answers is not in the document]
Comment (brasmusson) [reply gherkin/c/include/error_event.h:1]: As C code is distributed in source and is compiled with the users compiler - which we do not know which it is - portability is a very important property for open source code. I absolutely do not want to exclude users that are stack on old (what many people would call "obsolete") compilers.

[the remark this answers is not in the document]
Comment (aurelien-reeves) [reply gherkin/c/include/error_event.h:1]: Fine, that's legit

File: gherkin/c/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aurelien-reeves) [reply gherkin/c/README.md:74]: Rendered where?

I have no issue in the github UI
Are you using something else to look at the readme?

**Pull request merged**